i3lock-fancy

```diff
@@ -82,12 +85,13 @@ while true ; do
     esac
 done
 
-if "$shot_custom" && [[ $# -gt 0 ]]; then
-    shot=("$@");
+if [ -n "$shot" ] ; then
```

Comment:
Use `[[` like the rest of the script

Reply:
My direction will probably be to offer a re-do of the script using sh instead of bash, which should improve the script's response time, so I'll keep my copy the way it is. Feel free to change your version.

Reply:
If you have a look at the issue "script is slow" you can see all of the things people have already tried in optimizing the script. Currently it uses some bash specific things which is why its not an sh script. While some savings could be made in not having to load a bash process to run, there would be a loss in functionality. The majority of the script run time is taken up by image processing. I actually previously broke the script up in parts and tested run times for things when optimizing stuff and image magick was by far the longest running part of the script.

I have been considering possibly rewriting in lua which has nice pattern matching and would avoid all the callouts to awk, but my health is not good so I haven't been able to try writing something new from scratch to see if it would make any kind of impact.

Reply:
I agree changing to sh doesn't change anything besides restrict yourself to a subset of bash features.  Not sure why you're so stubborn about this @Boruch-Baum.